packages/shared/src/typeUtils.ts

```diff
@@ -21,3 +21,9 @@ export type Awaited<T> = T extends null | undefined
     ? Awaited<V> // recursively unwrap the value
     : never // the argument to `then` was not callable
   : T // non-object or non-thenable
+
+export type IsSameType<T, U> = T extends U
```

Comment:
Is it really necessary? It seems like it didn't make code less or pretty.

Reply:
Yeah, I think it will be frequently used. And it also exists in [the type definition of vue2](https://github.com/vuejs/vue/blob/main/types/common.d.ts).